driver/compile.ml

```diff
@@ -23,17 +23,18 @@ let interface = Compile_common.interface ~tool_name
 (** Bytecode compilation backend for .ml files. *)
 
 let to_bytecode i (typedtree, coercion) =
+  Compmisc.with_ppf_dump ~file_prefix:(Compile_common.impl i) @@ fun ppf_dump ->
```

Comment:
When printing to a file, `Compmisc.with_ppf_dump` truncates said file, so this change looks like it'd break -dump-into-file or whatever the option ended up being called.